src/RoboSaverVirtualModule.sol

```diff
@@ -35,147 +47,146 @@ contract RoboSaverVirtualModule {
                                        ERRORS
     //////////////////////////////////////////////////////////////////////////*/
 
-    error NotTopupAgent(address agent);
+    error NotKeeper(address agent);
 
     /*//////////////////////////////////////////////////////////////////////////
                                        EVENTS
     //////////////////////////////////////////////////////////////////////////*/
 
-    event SafeTopup(address indexed safe, uint256 amount, uint256 timestamp);
-    event BptTopup(address indexed safe, uint256 amount, uint256 timestamp);
+    event PoolWithdrawal(address indexed safe, uint256 amount, uint256 timestamp);
+    event PoolDeposit(address indexed safe, uint256 amount, uint256 timestamp);
 
     /*//////////////////////////////////////////////////////////////////////////
                                      CONSTRUCTOR
     //////////////////////////////////////////////////////////////////////////*/
-    constructor(address _delayModule, address _rolesModule, address _topupAgent, uint256 _eureBuffer) {
+
+    constructor(address _delayModule, address _rolesModule, address _keeper, uint256 _buffer) {
         delayModule = IDelayModifier(_delayModule);
         rolesModule = IRolesModifier(_rolesModule);
-
-        topupAgent = _topupAgent;
-
-        eureBuffer = _eureBuffer;
+        keeper = _keeper;
+        buffer = _buffer;
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                                       MODIFIERS
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @notice Checks whether a call is authorized to trigger top-up or exec queue txs
-    modifier onlyTopupAgents() {
-        if (msg.sender != topupAgent) revert NotTopupAgent(msg.sender);
+    modifier onlyKeeper() {
+        if (msg.sender != keeper) revert NotKeeper(msg.sender);
         _;
     }
 
+    /*//////////////////////////////////////////////////////////////////////////
+                    EXTERNAL METHODS: TOP-UP AGENTS
+    //////////////////////////////////////////////////////////////////////////*/
+
     /// @dev Check condition and determine whether a task should be executed by Gelato.
     function checker() external view returns (bool canExec, bytes memory execPayload) {
-        address cachedAvatar = delayModule.avatar();
-
-        uint256 queueNonce = delayModule.queueNonce();
-        uint256 txNonce = delayModule.txNonce();
-
-        if (txNonce != queueNonce) {
-            uint256 txQueuedAt = delayModule.getTxCreatedAt(queueNonce - 1);
-            // @note triggers the condition for exec the pendant tx in the delay module
-            if (block.timestamp - txQueuedAt >= delayModule.txCooldown()) {
-                return (true, abi.encodeWithSelector(IDelayModifier.executeNextTx.selector));
-            }
-
-            return (false, bytes("Tx cooldown not reached"));
```

Comment:
this logic should remain afaik!!!

Reply:
this logic will be handle on the gelato side while listening the event and building the right payload for `executeNextTx`. not need to track imo this `executeNextTx` in this piece of logic
 
maybe something to actually think about is if we would like to queue a tx if there is one pending or how to handle that. say, there is an existing transfer before which could bring you below the buffer etc